.emacs.d/lisp/code/code-navigation.el

```diff
@@ -0,0 +1,12 @@
+;;; M-n: go the next occurence of the symbol under point
+;;; M-p: go the previous occurence of the symbol under point
+;;;   note: these keybindings are enabled as a part of `highlight-symbol-nav-mode'
+;;; M-': replace all following occurences of the symbol under point
+
+;; enable highlight-symbol mode for all programming modes
+(add-hook 'prog-mode-hook (lambda ()
+                            (highlight-symbol-mode +1)
+                            (highlight-symbol-nav-mode +1)
```

Comment:
Yes. Modes enable on positive arguments and disable on non-positive arguments. To make this clear, it's a common practice to enable using +1 and disable using -1.